packages/server-core/src/media/storageprovider/ipfs.storage.ts

```diff
@@ -37,6 +38,16 @@ export class IPFSStorage implements StorageProviderInterface {
       .then((res) => res.type === 'directory')
       .catch(() => false)
   }
+  async getCachedAsset(assetPath: string): Promise<string> {
+    if (!this.cacheDomain) throw new Error('No cache domain found - please check the storage provider configuration')
+
+    const filePath = path.join(this._pathPrefix, assetPath)
+
+    return this._client.files
```

Comment:
we should use async here rather than returning the promise that has then and catch attached

also, why is it necessary that this uses the stat function? we should keep this synchronous and we do not need to check that the file exists, it is just a helper file to return a URL, we do not want to mutate the path if it cannot be found

Reply:
Well, I am using promise which in itself resolves or reject just like things have been done in `local.storage.ts`

I am making use of stats method to get `CID` which is used to access resources in IPFS MFS